File: Data Structures & Algorithms/minimum-stack/submission-1.cpp
class MinStack {
private:
std::stack<int> s;
std::stack<int> min_s;

public:
    MinStack() {
        
    }
    
    void push(int val) {
        s.push(val);
        if(min_s.empty() || val<=min_s.top()){
            min_s.push(val);
        }
    }
    
    void pop() {
        if(s.top() == min_s.top()) min_s.pop();
        s.pop();
    }
    
    int top() {
        return s.top();
        
    }
    
    int getMin() {
        return min_s.top();
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        dic = {'(':')', '{':'}', '[':']'}
        stack = []
        for c in s:
            if c in dic:
                stack.append(c)
            else:
                if not stack:
                    return False
                l = stack.pop()
                if c==dic[l]:
                    pass
                else:
                    return False
        return not stack

            

